### DISCUSSION TITLE
Reload the strongswan.conf configuration file during runtime

### DISCUSSION THREAD

**Discussion by nknkgithub** [strongswan/strongswan#2188] [General]
Is there a functionality similar to swanctl --reload-settings in ipsec?
Does "ipsec update" reload strongswan.conf file during runtime?

We are using ipsec.conf and our objective is to enable dynamic configuration for performing OCSP.

**Comment by strongX509** [comment]
If you are still using ipsec.conf and the deprecated stroke interface then the only way is to send a SIGHUP signal to the charon daemon in order to reload the strongswan.conf settings ([see the documentation on strongswan.conf](https://docs.strongswan.org/docs/5.9/config/strongswanConf.html#_reloading))

**Comment by nknkgithub** [comment]
Thanks @strongX509 .

We tried sending SIGHUP signal to charon daemon and below are our observations

a) In strongswan.conf enable_ocsp is set to true, OCSP check is performed and tunnel is up.
```
revocation{
                       enable_ocsp = yes
                       enable_crl = no
                   }
```
b) We want to disable OCSP check. Hence updated enable_ocsp = no in strongswan.conf
```
revocation{
                       enable_ocsp = no
                       enable_crl = no
                   }
```
c) Sent SIGHUP signal to charon
```
ps -alef | grep -i charon
0 S root      4227  3905  0  80   0 -   980 pipe_w 06:53 pts/3    00:00:00 grep -i charon
1 S root      9200     1  0  80   0 -   746 poll_s Apr01 ?        00:00:00 /usr/lib/strongswan/ipsec/starter --daemon charon
4 S root      9201  9200  0  80   0 - 42672 sigtim Apr01 ?        00:00:02 /usr/lib/strongswan/ipsec/charon --use-syslog --debug-ike 0 --debug-chd 1 --debug-cfg 1 --debug-net 1 --debug-enc 1 --debug-lib 1 --debug-mgr 1 --debug-knl 1 --debug-dmn -1

kill -HUP 9201
```

d) Brought down the tunnel.
`ipsec down home3`

e) Brought up tunnel. Observing OCSP check is done and updated configuration is not taken into account. 
`ipsec up home3`
```
received end entity cert "C=US, O=<O>, OU=<OU>, CN=<End entity cert>"
  using certificate "C=US, O=<O>, OU=<OU>, CN=<End entity cert>"
  using trusted ca certificate "C=US, O=<O>, OU=<OU>, CN=<Root CA>"
checking certificate status of "C=US, O=<O>, OU=<OU>, CN=<End entity cert>"
  ocsp response correctly signed by "C=US, O=<O>, OU=<OU>, CN=<Signer Cert>"
  ocsp response contains no status on our certificate
  reached self-signed root ca with a path length of 0
authentication of 'C=US, O=<O>, OU=<OU>, CN=<End entity cert>' with RSA_EMSA_PKCS1_SHA2_256 successful
```

ipsec version
```
ipsec version
Linux strongSwan U5.6.3/K4.1.35-rt41
Institute for Internet Technologies and Applications
University of Applied Sciences Rapperswil, Switzerland
See 'ipsec --copyright' for copyright information.
```


Can you please help?

**Reply by tobiasbrunner** [reply]
Your version does not support changing these options at runtime. That was added with 5.7.0.
